OpenEdXMobile/src/main/java/org/edx/mobile/view/MainDiscoveryFragment.java

```diff
@@ -171,12 +169,19 @@ public void setUserVisibleHint(boolean isVisibleToUser) {
         onFragmentVisibilityChanged(isVisibleToUser);
     }
 
-    public void onFragmentVisibilityChanged(boolean isVisibleToUser) {
+    public void onFragmentVisibilityChanged(final boolean isVisibleToUser) {
         if (toolbarCallbacks != null && toolbarCallbacks.getSearchView() != null) {
-            toolbarCallbacks.getSearchView().setVisibility(isVisibleToUser ? View.VISIBLE : View.GONE);
-        }
-        if (shownFragment != null && shownFragment instanceof NativeFindCoursesFragment) {
-            toolbarCallbacks.getSearchView().setVisibility(View.GONE);
+            toolbarCallbacks.getSearchView().post(new Runnable() {
```

Comment:
This should have an accompanying comment as to why this piece of code was added in the Runnable.